### ISSUE TITLE
bwrap can't bind mount

### ISSUE THREAD

**Issue by ehENTER** [flathub/com.valvesoftware.Steam#44]
OS: Void Linux x86_64-musl
Kernel version: 4.14.4
Flatpak version: 0.10.1
Bubblewrap version: 0.2.0

This happens when trying to run Steam:
```
> flatpak run com.valvesoftware.Steam
bwrap: Can't bind mount /oldroot/home/ehenter/.var/app/com.valvesoftware.Steam on /newroot/home/ehenter/.: Invalid argument
```

Steam version:
```
> flatpak info com.valvesoftware.Steam
Ref: app/com.valvesoftware.Steam/x86_64/stable
ID: com.valvesoftware.Steam
Arch: x86_64
Branch: stable
Origin: flathub
Date: 2017-10-31 13:36:51 +0000
Subject: Build com.valvesoftware.Steam at fb26150f57a47e748a36c7a757eb364aa654233a
Commit: 93dacf895d8d4e1c78c34780cefbea7cfcb1a69f460ce5029b27e23a901f7d96
Parent: 76a55c2031546228deb58f4c0fe62d497ff5babc05ce8474145ca1c8e01bb217
Location: /var/lib/flatpak/app/com.valvesoftware.Steam/x86_64/stable/93dacf895d8d4e1c78c34780cefbea7cfcb1a69f460ce5029b27e23a901f7d96
Installed size: 2.9 MB
Runtime: org.freedesktop.Platform/x86_64/1.6
```
I don't have that kind of problems when using other flatpak apps. I have tried to reinstall com.valvesoftware.Steam and I have tried to install it with --user flag. But this problem persits.

**Comment by E100Beta** [comment]
I have the same problem.
My setup is mostly the same as OP:

---
**note:** I used `-vv` in `flatpak run`, maybe it'll help.

OS: Void Linux x86_64-musl
Kernel version: 4.14.6
Flatpak version: 0.10.1
Bubblewrap version: 0.2.0

When i do:

    flatpak run -vv com.valvesoftware.Steam

I get:
```
F: Opening user flatpak installation at path /home/pavel/.local/share/flatpak
F: No installations directory in /etc/flatpak/installations.d. Skipping
F: Opening user flatpak installation at path /home/pavel/.local/share/flatpak
F: Allowing x11 access
F: Allowing pulseaudio access
F: Allowing system-dbus access
F: Failed to run in transient scope: No systemd user session available, cgroups not available
F: Running '--ro-bind /home/pavel/.local/share/flatpak/runtime/org.freedesktop.Platform/x86_64/1.6/d226d18194f2349ffef10fce612910b0a2d761883a060e78b4087c21f32790fd/files /usr --lock-file /usr/.ref --ro-bind /home/pavel/.local/share/flatpak/app/com.valvesoftware.Steam/x86_64/stable/4819a16e6494c41dfd2901432a91a6c73177bc3f6c708a9059e90325ae0ab0be/files /app --lock-file /app/.ref --ro-bind /home/pavel/.local/share/flatpak/runtime/org.freedesktop.Platform.Compat32/x86_64/1.6/ac77e92a27f418e44847316d94724bf5f1f9d96e10a02be9eb6911fedacbfddf/files /app/lib/32bit --ro-bind /home/pavel/.local/share/flatpak/runtime/org.freedesktop.Platform.VAAPI.Intel/x86_64/1.6/5962b0bb899694972d2050fc9f726b5215231dcd8264ee1e55bef3c53508caf7/files /usr/lib/dri/intel-vaapi-driver --ro-bind /home/pavel/.local/share/flatpak/runtime/org.freedesktop.Platform.Locale/x86_64/1.6/fc3c1ef6d7b24c0079af06f1ea8e255ba58972fdfce8f2c1f9c9a20174fffb8a-en/files /usr/share/runtime/locale --unshare-pid --proc /proc --dir /tmp --dir /var/tmp --dir /run/host --dir /run/user/1000 --setenv XDG_RUNTIME_DIR /run/user/1000 --symlink ../run /var/run --ro-bind /sys/block /sys/block --ro-bind /sys/bus /sys/bus --ro-bind /sys/class /sys/class --ro-bind /sys/dev /sys/dev --ro-bind /sys/devices /sys/devices --bind-data 4 /etc/passwd --bind-data 5 /etc/group --ro-bind /var/lib/dbus/machine-id /etc/machine-id --bind /home/pavel/.local/share/flatpak/runtime/org.freedesktop.Platform/x86_64/1.6/d226d18194f2349ffef10fce612910b0a2d761883a060e78b4087c21f32790fd/files/etc/fstab /etc/fstab --bind /home/pavel/.local/share/flatpak/runtime/org.freedesktop.Platform/x86_64/1.6/d226d18194f2349ffef10fce612910b0a2d761883a060e78b4087c21f32790fd/files/etc/default /etc/default --bind /home/pavel/.local/share/flatpak/runtime/org.freedesktop.Platform/x86_64/1.6/d226d18194f2349ffef10fce612910b0a2d761883a060e78b4087c21f32790fd/files/etc/geoclue /etc/geoclue --bind /home/pavel/.local/share/flatpak/runtime/org.freedesktop.Platform/x86_64/1.6/d226d18194f2349ffef10fce612910b0a2d761883a060e78b4087c21f32790fd/files/etc/timezone /etc/timezone --bind /home/pavel/.local/share/flatpak/runtime/org.freedesktop.Platform/x86_64/1.6/d226d18194f2349ffef10fce612910b0a2d761883a060e78b4087c21f32790fd/files/etc/profile /etc/profile --bind /home/pavel/.local/share/flatpak/runtime/org.freedesktop.Platform/x86_64/1.6/d226d18194f2349ffef10fce612910b0a2d761883a060e78b4087c21f32790fd/files/etc/skel /etc/skel --bind /home/pavel/.local/share/flatpak/runtime/org.freedesktop.Platform/x86_64/1.6/d226d18194f2349ffef10fce612910b0a2d761883a060e78b4087c21f32790fd/files/etc/inputrc /etc/inputrc --bind /home/pavel/.local/share/flatpak/runtime/org.freedesktop.Platform/x86_64/1.6/d226d18194f2349ffef10fce612910b0a2d761883a060e78b4087c21f32790fd/files/etc/xml /etc/xml --bind /home/pavel/.local/share/flatpak/runtime/org.freedesktop.Platform/x86_64/1.6/d226d18194f2349ffef10fce612910b0a2d761883a060e78b4087c21f32790fd/files/etc/ca-certificates.conf /etc/ca-certificates.conf --bind /home/pavel/.local/share/flatpak/runtime/org.freedesktop.Platform/x86_64/1.6/d226d18194f2349ffef10fce612910b0a2d761883a060e78b4087c21f32790fd/files/etc/nsswitch.conf /etc/nsswitch.conf --bind /home/pavel/.local/share/flatpak/runtime/org.freedesktop.Platform/x86_64/1.6/d226d18194f2349ffef10fce612910b0a2d761883a060e78b4087c21f32790fd/files/etc/ld.so.conf /etc/ld.so.conf --bind /home/pavel/.local/share/flatpak/runtime/org.freedesktop.Platform/x86_64/1.6/d226d18194f2349ffef10fce612910b0a2d761883a060e78b4087c21f32790fd/files/etc/issue /etc/issue --bind /home/pavel/.local/share/flatpak/runtime/org.freedesktop.Platform/x86_64/1.6/d226d18194f2349ffef10fce612910b0a2d761883a060e78b4087c21f32790fd/files/etc/version /etc/version --bind /home/pavel/.local/share/flatpak/runtime/org.freedesktop.Platform/x86_64/1.6/d226d18194f2349ffef10fce612910b0a2d761883a060e78b4087c21f32790fd/files/etc/rpc /etc/rpc --bind /home/pavel/.local/share/flatpak/runtime/org.freedesktop.Platform/x86_64/1.6/d226d18194f2349ffef10fce612910b0a2d761883a060e78b4087c21f32790fd/files/etc/init.d /etc/init.d --bind /home/pavel/.local/share/flatpak/runtime/org.freedesktop.Platform/x86_64/1.6/d226d18194f2349ffef10fce612910b0a2d761883a060e78b4087c21f32790fd/files/etc/pki /etc/pki --bind /home/pavel/.local/share/flatpak/runtime/org.freedesktop.Platform/x86_64/1.6/d226d18194f2349ffef10fce612910b0a2d761883a060e78b4087c21f32790fd/files/etc/protocols /etc/protocols --bind /home/pavel/.local/share/flatpak/runtime/org.freedesktop.Platform/x86_64/1.6/d226d18194f2349ffef10fce612910b0a2d761883a060e78b4087c21f32790fd/files/etc/dbus-1 /etc/dbus-1 --bind /home/pavel/.local/share/flatpak/runtime/org.freedesktop.Platform/x86_64/1.6/d226d18194f2349ffef10fce612910b0a2d761883a060e78b4087c21f32790fd/files/etc/vdpau_wrapper.cfg /etc/vdpau_wrapper.cfg --bind /home/pavel/.local/share/flatpak/runtime/org.freedesktop.Platform/x86_64/1.6/d226d18194f2349ffef10fce612910b0a2d761883a060e78b4087c21f32790fd/files/etc/ld.so.cache /etc/ld.so.cache --bind /home/pavel/.local/share/flatpak/runtime/org.freedesktop.Platform/x86_64/1.6/d226d18194f2349ffef10fce612910b0a2d761883a060e78b4087c21f32790fd/files/etc/terminfo /etc/terminfo --bind /home/pavel/.local/share/flatpak/runtime/org.freedesktop.Platform/x86_64/1.6/d226d18194f2349ffef10fce612910b0a2d761883a060e78b4087c21f32790fd/files/etc/ca-certificates /etc/ca-certificates --bind /home/pavel/.local/share/flatpak/runtime/org.freedesktop.Platform/x86_64/1.6/d226d18194f2349ffef10fce612910b0a2d761883a060e78b4087c21f32790fd/files/etc/shells /etc/shells --bind /home/pavel/.local/share/flatpak/runtime/org.freedesktop.Platform/x86_64/1.6/d226d18194f2349ffef10fce612910b0a2d761883a060e78b4087c21f32790fd/files/etc/pkcs11 /etc/pkcs11 --bind /home/pavel/.local/share/flatpak/runtime/org.freedesktop.Platform/x86_64/1.6/d226d18194f2349ffef10fce612910b0a2d761883a060e78b4087c21f32790fd/files/etc/sysctl.conf /etc/sysctl.conf --bind /home/pavel/.local/share/flatpak/runtime/org.freedesktop.Platform/x86_64/1.6/d226d18194f2349ffef10fce612910b0a2d761883a060e78b4087c21f32790fd/files/etc/fonts /etc/fonts --symlink /proc/mounts /etc/mtab --bind /home/pavel/.local/share/flatpak/runtime/org.freedesktop.Platform/x86_64/1.6/d226d18194f2349ffef10fce612910b0a2d761883a060e78b4087c21f32790fd/files/etc/xdg /etc/xdg --bind /home/pavel/.local/share/flatpak/runtime/org.freedesktop.Platform/x86_64/1.6/d226d18194f2349ffef10fce612910b0a2d761883a060e78b4087c21f32790fd/files/etc/rpm /etc/rpm --bind /home/pavel/.local/share/flatpak/runtime/org.freedesktop.Platform/x86_64/1.6/d226d18194f2349ffef10fce612910b0a2d761883a060e78b4087c21f32790fd/files/etc/motd /etc/motd --bind /home/pavel/.local/share/flatpak/runtime/org.freedesktop.Platform/x86_64/1.6/d226d18194f2349ffef10fce612910b0a2d761883a060e78b4087c21f32790fd/files/etc/ssl /etc/ssl --bind /home/pavel/.local/share/flatpak/runtime/org.freedesktop.Platform/x86_64/1.6/d226d18194f2349ffef10fce612910b0a2d761883a060e78b4087c21f32790fd/files/etc/gtk-3.0 /etc/gtk-3.0 --bind /home/pavel/.local/share/flatpak/runtime/org.freedesktop.Platform/x86_64/1.6/d226d18194f2349ffef10fce612910b0a2d761883a060e78b4087c21f32790fd/files/etc/services /etc/services --symlink /usr/lib/GL/glvnd /etc/glvnd --bind /home/pavel/.local/share/flatpak/runtime/org.freedesktop.Platform/x86_64/1.6/d226d18194f2349ffef10fce612910b0a2d761883a060e78b4087c21f32790fd/files/etc/drirc /etc/drirc --bind /home/pavel/.local/share/flatpak/runtime/org.freedesktop.Platform/x86_64/1.6/d226d18194f2349ffef10fce612910b0a2d761883a060e78b4087c21f32790fd/files/etc/filesystems /etc/filesystems --bind /home/pavel/.local/share/flatpak/runtime/org.freedesktop.Platform/x86_64/1.6/d226d18194f2349ffef10fce612910b0a2d761883a060e78b4087c21f32790fd/files/etc/issue.net /etc/issue.net --bind /home/pavel/.local/share/flatpak/runtime/org.freedesktop.Platform/x86_64/1.6/d226d18194f2349ffef10fce612910b0a2d761883a060e78b4087c21f32790fd/files/etc/rcS.d /etc/rcS.d --symlink ../usr/lib/os-release /etc/os-release --symlink /usr/lib/GL/vulkan /etc/vulkan --bind /home/pavel/.local/share/flatpak/runtime/org.freedesktop.Platform/x86_64/1.6/d226d18194f2349ffef10fce612910b0a2d761883a060e78b4087c21f32790fd/files/etc/locale.conf /etc/locale.conf --bind /home/pavel/.local/share/flatpak/runtime/org.freedesktop.Platform/x86_64/1.6/d226d18194f2349ffef10fce612910b0a2d761883a060e78b4087c21f32790fd/files/etc/hostname /etc/hostname --bind /home/pavel/.local/share/flatpak/runtime/org.freedesktop.Platform/x86_64/1.6/d226d18194f2349ffef10fce612910b0a2d761883a060e78b4087c21f32790fd/files/etc/timestamp /etc/timestamp --bind /home/pavel/.local/share/flatpak/runtime/org.freedesktop.Platform/x86_64/1.6/d226d18194f2349ffef10fce612910b0a2d761883a060e78b4087c21f32790fd/files/etc/pulse /etc/pulse --bind /home/pavel/.var/app/com.valvesoftware.Steam/cache /var/cache --bind /home/pavel/.var/app/com.valvesoftware.Steam/data /var/data --bind /home/pavel/.var/app/com.valvesoftware.Steam/config /var/config --bind /home/pavel/.var/app/com.valvesoftware.Steam/cache/tmp /var/tmp --symlink usr/lib /lib --symlink usr/lib64 /lib64 --symlink usr/bin /bin --symlink usr/sbin /sbin --seccomp 6 --ro-bind /home/pavel/.cache/flatpak-monitor /run/host/monitor --symlink /run/host/monitor/localtime /etc/localtime --symlink /run/host/monitor/resolv.conf /etc/resolv.conf --symlink /run/host/monitor/host.conf /etc/host.conf --symlink /run/host/monitor/hosts /etc/hosts --bind-data 12 /etc/ld.so.conf --ro-bind-data 3 /etc/ld.so.cache --file 9 /.flatpak-info --ro-bind-data 11 /.flatpak-info --symlink ../../../.flatpak-info /run/user/1000/flatpak-info --bind /home/pavel/.cache/doc/by-app/com.valvesoftware.Steam /run/user/1000/doc --dev-bind /dev /dev --bind /home/pavel/.var/app/com.valvesoftware.Steam/. /home/pavel/. --bind /home/pavel/.cache/app/com.valvesoftware.Steam /run/user/1000/app/com.valvesoftware.Steam --dir /home/pavel --dir /home/pavel/.local/share/flatpak --dir /home/pavel/.var/app --bind /home/pavel/.var/app/com.valvesoftware.Steam /home/pavel/.var/app/com.valvesoftware.Steam --tmpfs /tmp/.X11-unix --bind /tmp/.X11-unix/X0 /tmp/.X11-unix/X99 --bind-data 14 /run/user/1000/Xauthority --ro-bind /usr/share/fonts /run/host/fonts --ro-bind /var/cache/fontconfig /run/host/fonts-cache --ro-bind /home/pavel/.cache/fontconfig /run/host/user-fonts-cache --ro-bind /usr/share/icons /run/host/share/icons --symlink /app/lib/debug/source /run/build --symlink /usr/lib/debug/source /run/build-runtime steam-wrapper'
bwrap: Can't bind mount /oldroot/home/pavel/.var/app/com.valvesoftware.Steam on /newroot/home/pavel/.: Invalid argument
```

Steam version:
```
Ref: app/com.valvesoftware.Steam/x86_64/stable
ID: com.valvesoftware.Steam
Arch: x86_64
Branch: stable
Origin: flathub
Date: 2017-12-15 14:25:55 +0000
Subject: Update email and add a longer description (4fce1feb)
Commit: 4819a16e6494c41dfd2901432a91a6c73177bc3f6c708a9059e90325ae0ab0be
Parent: 1c298700ad50859facc4fccbdfe0154860c39cc35c5a09fab46844438134d397
Location: /home/pavel/.local/share/flatpak/app/com.valvesoftware.Steam/x86_64/stable/4819a16e6494c41dfd2901432a91a6c73177bc3f6c708a9059e90325ae0ab0be
Installed size: 6.3 MB
Runtime: org.freedesktop.Platform/x86_64/1.6
```

---

I installed it with --user flag.

So far I use Archlinux in a chroot to play steam games, would be very nice if I could use a more more convenient approach.

**Comment by TingPing** [comment]
This might be an issue worth reporting against flatpak/flatpak as I don't think its specific to this application.

**Comment by E100Beta** [comment]
@TingPing as a test, I installed Linphone (it's first on flatpak.org/apps and is also distributed on flathub), and it worked fine. Then I picked Libreoffice at random, and it too ran without any problems.

So I suspect it **is** application-specific.

**Comment by TingPing** [comment]
@E100Beta Ah the key here is this package uses `"--persist=."` so I'm guessing you have an old version of `flatpak` and upgrading would probably fix that.

**Comment by E100Beta** [comment]
Thanks for giving me a direction to work in!
I manually compiled and installed flatpak 0.10.2. Unfortunately, that didn't change anything.

Can you think of anything I can do on my end to mitigate the issue? I tried to pass different arguments to `flatpak run` and `override` but I don't really know what to do.

**Comment by ehENTER** [comment]
I just tested `com.jagex.RuneScape` with `flatpak override --persist=. com.jagex.RuneScape` and then same error message pops up as with Steam when trying to run it. So it seems to be problem with `flatpak` itself. And also I just tested `com.valvesoftware.Steam` on my another computer which runs up-to-date Void Linux with glibc and it works just fine. So this just might be musl specific issue. Should I report this on flatpak/flatpak?
